torchtune/dev/rl/generation.py

```diff
@@ -0,0 +1,244 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
```

Comment:
How much of generation.py, gsm8k.py, data.py, rewards.py, types.py is identical to what's already in dev/grpo? (At least for this file they seem identical.) Obv not for this PR but we should consolidate

Reply:
Rewards is quite different, the rest are closer. We can def consolidate down the road...